runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -345,26 +366,18 @@ public void removeFromStripe(String endpoint, LayoutStripe layoutStripe,
      */
     public LayoutBuilder removeLogunitServer(String endpoint) {
 
-        List<LayoutSegment> layoutSegments = layout.getSegments();
+        Layout tempLayout = new Layout(layout);
+
+        List<LayoutSegment> layoutSegments = tempLayout.getSegments();
         for (LayoutSegment layoutSegment : layoutSegments) {
+            Layout.ReplicationMode mode = layoutSegment.getReplicationMode();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "mode". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "mode" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)